src/qt/sendcoinsdialog.cpp

```diff
@@ -399,15 +404,51 @@ void SendCoinsDialog::on_sendButton_clicked()
         CDataStream ssTx(SER_NETWORK, PROTOCOL_VERSION);
         ssTx << psbtx;
         GUIUtil::setClipboard(EncodeBase64(ssTx.str()).c_str());
-        Q_EMIT message(tr("PSBT copied"), "Copied to clipboard", CClientUIInterface::MSG_INFORMATION);
+        QMessageBox msgBox;
+        msgBox.setText("Unsigned Transaction");
+        msgBox.setInformativeText("The PSBT has been copied to the clipboard. You can also save it.");
+        msgBox.setStandardButtons(QMessageBox::Save | QMessageBox::Discard);
+        msgBox.setDefaultButton(QMessageBox::Discard);
+        switch (msgBox.exec()) {
+        case QMessageBox::Save: {
+            QString selectedFilter;
+            QString fileNameSuggestion = "";
+            bool first = true;
+            for (const SendCoinsRecipient &rcp : m_current_transaction->getRecipients()) {
+                if (!first) {
+                    fileNameSuggestion.append("-");
+                }
+                QString amount = BitcoinUnits::format(model->getOptionsModel()->getDisplayUnit(), rcp.amount);
+                fileNameSuggestion.append(rcp.address + "-" + amount);
```

Comment:
For better UX, we should probably use the recipient label if it is available and truncate the 0's on the amount.

Reply:
Added the recipient name and switched to `formatWithUnit`. We could add an argument to `BitcoinUnits::format` to remove trailing zeros ... in a later PR.